libs/LzmaLib/source/MtCoder.c

```diff
@@ -355,7 +355,12 @@ static THREAD_FUNC_RET_TYPE THREAD_FUNC_CALL_TYPE ThreadFunc(void *pp)
       
       #ifndef MTCODER__USE_WRITE_THREAD
       {
+// InterlockedIncrement is defined in winnt.h, not available on POSIX systems
+#ifdef _WIN32
         unsigned numFinished = (unsigned)InterlockedIncrement(&mtc->numFinishedThreads);
+#else
+        unsigned numFinished = (unsigned)__sync_fetch_and_add(&mtc->numFinishedThreads, 1);
```

Comment:
According to this stack overflow entry: https://stackoverflow.com/questions/2125937/equivalent-of-interlockedincrement-in-linux-gcc
This should be the equivalent :)
```suggestion
        unsigned numFinished = (unsigned)__sync_add_and_fetch(&mtc->numFinishedThreads, 1);
```

Reply:
Good spot! I thought I grabbed the correct one from that answer, clearly I didn't!